src/verifast1.ml

```diff
@@ -3217,34 +3337,44 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
         Static ->
         begin fun on_fail ->
           match try_assoc "this" tenv with
-            Some (ObjType cn) ->
-            try_qualified_call cn (Var (l, "this")::es) es Instance on_fail
+            Some (ObjType (cn, targs)) ->
+            try_qualified_call cn (Var (l, "this")::es) es Instance targs on_fail

```

Comment:
You treat receiver type arguments as if they were method type arguments. It seems better to distinguish them. In particular: if the receiver is a raw type, this is not the same as calling the method without specifying any type arguments. It is not okay to infer the type arguments in that case. It is unsound if VeriFast infers a different type than the Java compiler does.

Reply:
In other words: distinguish class type parameters and method type parameters.